src/main/java/com/google/javascript/cl2dts/DeclarationGenerator.java

```diff
@@ -490,6 +496,8 @@ public Void caseStringType() {
         public Void caseObjectType(ObjectType type) {
           if (type.isRecordType()) {
             visitRecordType((RecordType) type);
+          } else if (type.isDict()) {
```

Comment:
yes, I added a test case for that, and the closure compiler is happy to produce such a thing.
given the lack of a spec, that means it Must Be Valid Closure